lighthouse-core/gather/driver.js

```diff
@@ -506,12 +557,19 @@ class Driver {
       options: 'sampling-frequency=10000'  // 1000 is default and too slow.
     };
 
-    // Disable any domains that could interfere or add overhead to the trace
-    return this.sendCommand('Debugger.disable')
-      .then(_ => this.sendCommand('CSS.disable'))
-      .then(_ => this.sendCommand('DOM.disable'))
-      // Enable Page domain to wait for Page.loadEventFired
-      .then(_ => this.sendCommand('Page.enable'))
+    // Check any domains that could interfere with or add overhead to the trace.
+    if (this.domainEnabled('Debugger')) {
```

Comment:
this moves these to just being a log error message. Disabling these domains after `beforePass` has executed if a trace is being taken means that e.g. if someone adds a DBW gatherer that needs a trace that any DBW gatherer that enabled `Debugger`, `CSS`, or `DOM` in `beforePass` will be broken, possibly fairly subtly. Better to warn on these...or even throw an `Error`?